pkg/templates/generic_lookup_func.go

```diff
@@ -259,3 +280,23 @@ func (t *TemplateResolver) discoverAPIResources() ([]*metav1.APIResourceList, er
 
 	return apiresourcelist, nil
 }
+
+func onAllowlist(allowlist []ClusterScopedObjectIdentifier, rsrc ClusterScopedObjectIdentifier) bool {
+	if len(allowlist) == 0 {
+		return false
+	}
+
+	for _, item := range allowlist {
+		if item.Group == "*" && item.Version == "*" && item.Kind == "*" && item.Name == "*" {
```

Comment:
I think something like this would be better:

```go
if item.Group != "*" && item.Group != rsrc.Group {
continue
}

if item.Kind != "*" && item.Kind != rsrc.Kind {
continue
}

if item.Name == "*" || item.Name == rsrc.Name {
return true
}
```